plugins/WaveShaper/WaveShaper.cpp

```diff
@@ -119,17 +124,17 @@ bool WaveShaperEffect::processAudioBuffer( sampleFrame * _buf,
 
 			if( lookup < 1 )
 			{
-				s[i] = frac * samples[0] * posneg;
+				s[i] = frac * graphSamples->operator[](0) * posneg;
 			}
 			else if( lookup < 200 )
 			{
-				s[i] = linearInterpolate( samples[ lookup - 1 ],
-						samples[ lookup ], frac )
+				s[i] = linearInterpolate(graphSamples->operator[](lookup - 1),
```

Comment:
Is there a reason you can't just do `(*graphSamples)[lookup - 1]` ?

Reply:
The type is std::vector<float>*, why use casting and use `[]` when `[]` can simply be used?

Reply:
There is no casting. The difference is just using normal array access method.
```cpp
graphSamples->operator[](lookup - 1)
(*graphSamples)[lookup -1]
```
This accomplishes the same thing. It's just odd seeing calling `->operator[]()` directly.

Reply:
So replace all `operator[]()`? It is your decision.

Reply:
I think `(*graphSamples)[lookup -1]` reads better.